File: test_security.py
import re

def test_check_password():
    # Valid password
    assert check_password("Valid123") is True
    # Too short
    assert check_password("Va1") is False
    # Missing uppercase
    assert check_password("valid123") is False
    # Missing lowercase
    assert check_password("VALID123") is False
    # Missing digit
    assert check_password("ValidPass") is False

def test_generate_uuid_v4():
    uuid_value = generate_uuid_v4()
    # Check format using regex (UUID v4 format)
    pattern = re.compile(
        r'^[a-f0-9]{8}-'
        r'[a-f0-9]{4}-'
        r'4[a-f0-9]{3}-'
        r'[89ab][a-f0-9]{3}-'
        r'[a-f0-9]{12}$'
    )
    assert pattern.match(uuid_value)

File: test_security.cs
// Author: Steve Sekapin Mnouze
// File: SecurityTests.cs

using Xunit;
using Assignment2;

namespace Assignment2Tests
{
    public class SecurityTests
    {
        [Fact]
        public void TestValidPassword()
        {
            Assert.True(Security.CheckPassword("Valid123!"));
        }

        [Fact]
        public void TestInvalidPasswords()
        {
            Assert.False(Security.CheckPassword("Va1"));           // too short
            Assert.False(Security.CheckPassword("valid123!"));      // no uppercase
            Assert.False(Security.CheckPassword("VALID123!"));      // no lowercase
            Assert.False(Security.CheckPassword("ValidPass!"));     // no digit
            Assert.False(Security.CheckPassword("Valid123"));       // no special char
        }

        [Fact]
        public void TestUUIDv4Format()
        {
            var uuid = Security.GenerateUUIDv4();
            // Regex for UUIDv4 format
            Assert.Matches(@"^[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-4[0-9a-fA-F]{3}\-[89abAB][0-9a-fA-F]{3}\-[0-9a-fA-F]{12}$", uuid);
        }
    }
}
